.github/workflows/e2e-test.yaml

```
@@ -0,0 +1,48 @@
+# Licensed to Apache Software Foundation (ASF) under one or more contributor
+# license agreements. See the NOTICE file distributed with
+# this work for additional information regarding copyright
+# ownership. Apache Software Foundation (ASF) licenses this file to you under
+# the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+# 
+#     http://www.apache.org/licenses/LICENSE-2.0
+# 
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+
+name: E2E Test
+
+on:
+  pull_request:
+
+jobs:
+  build:
+    name: E2E Test
+    runs-on: ubuntu-latest
+    steps:
+            
+      - name: Set up Go 1.18
+        uses: actions/setup-go@v4
+        with:
+          go-version: 1.18
+        id: go
+
+      - name: Check out code into the Go module directory
+        uses: actions/checkout@v3
+
+      - name: Build e2e
+        run: make linux
+
+      - name: Run E2E Test
+        uses: apache/skywalking-infra-e2e@main
```

This always run the codes on main branch, in order to run the codes in the PR, not on the main branch, maybe you should try something like:

```suggestion
        uses: .
```

---

`uses: apache/skywalking-infra-e2e@main` is just what we want, it's the external e2e, and note that we build a e2e using the PR code as the internal e2e (to be tested)

---

Oh right!